merkle/types/db.go

```diff
@@ -17,6 +23,72 @@ type TreeInfo struct {
 	Done bool `json:"done"`
 }
 
+func (t LegacyTreeInfo) Migrate(version uint64) TreeInfo {
+	return TreeInfo{
+		Version:        version,
+		Index:          t.Index,
+		LeafCount:      t.LeafCount,
+		StartLeafIndex: t.StartLeafIndex,
+		LastSiblings:   t.LastSiblings,
+		Done:           t.Done,
+	}
+}
+
+type TreeInfo struct {
+	// Version of the tree
+	Version uint64 `json:"version"`
+
+	// Index of the tree used as prefix for the keys
+	Index uint64 `json:"index"`
+
+	// Number of leaves in the tree
+	LeafCount uint64 `json:"leaf_count"`
+
+	// Cumulative number of leaves all the way up to the current tree
+	StartLeafIndex uint64 `json:"start_leaf_index"`
+
+	// Last sibling of the height(level) of the tree
+	LastSiblings map[uint8][]byte `json:"last_siblings"`
+
+	// Flag to indicate if the tree is finalized
+	Done bool `json:"done"`
+}
+
+func NewTreeInfo(version uint64, index uint64, leafCount uint64, startLeafIndex uint64, lastSiblings map[uint8][]byte, done bool) TreeInfo {
+	return TreeInfo{
+		Version:        version,
+		Index:          index,
+		LeafCount:      leafCount,
+		StartLeafIndex: startLeafIndex,
+		LastSiblings:   lastSiblings,
+		Done:           done,
+	}
+}
+
+func (t TreeInfo) Key() []byte {
+	return PrefixedWorkingTreeKey(t.Version)
```

Comment:
When I see the comments, it describes index is the prefix of the tree, but seems it is using version as prefix.